JetAnalyzers/bin/jet_response_fitter_x.cc

```diff
@@ -255,6 +271,584 @@ int main(int argc,char**argv)
+  TF1* modified_fdscb{0};
+
+  // fit definition
+  string gaus = "[0]*TMath::Exp(-0.5*TMath::Power((x-[1])/[2],2.))"; 
+  string expo = "[0]*TMath::Exp([1]*(x - [2]))";
+  string pol = "[0]+[1]*x+[2]*TMath::Power(x,2)+[3]*TMath::Power(x,3)+[4]*TMath::Power(x,4)+[5]*TMath::Power(x,5)+[6]*TMath::Power(x,6)+[7]*TMath::Power(x,7)";
+
+  // global fit maximun parameters and name
+  int maxPar{31};
+  vector<string> parName {"norm G1", "mean G1", "sigma G1 left", "norm Exp left", "slope Exp left", 
+    "p0 left", "p1 left", "p2 left", "p3 left", "p4 left", "p5 left", "p6 left", "p7 left", 
+    "R1", "R2", "sigma G1 right", "norm G2", "mean G2", "sigma G2", "norm G3", "mean G3", "sigma G3",
+    "p0 right", "p1 right", "p2 right", "p3 right", "p4 right", "p5 right", "p6 right", "p7 right", "xbinmax" };
+
+  // reduce chisquare and Kolmogorov-Smir test
+  double chi2, ks;
+  // map chi2/KS->fit
+  map<double, TH1F*> mapChi2Fit;
+  map<double, TH1F*> mapChi2FitB;
+  map<double, TH1F*> mapChi2FitO;
+  map<double, TH1F*> mapChi2FitM;
+  map<double, TH1F*> mapChi2FitG;
+  map<double, TH1F*> mapKsFit;  // the best out of the 4 cases
+  map<double, TH1F*> mapKsFitB; // best from Blue (case 4)
+  map<double, TH1F*> mapKsFitO; // best from Orange (case 3)
+  map<double, TH1F*> mapKsFitM; // best from Magenta(case 2)
+  map<double, TH1F*> mapKsFitG; // best from Green (case 1)
+
+  double xbinmax = hrsp->GetXaxis()->GetBinUpEdge(binmax); // x position of the max bin
+  double ybinmax = hrsp->GetBinContent(binmax);
+  //int bindown = binmax - 5;
+  int bindown = binmax - 4;
+  double minGaus = hrsp->GetXaxis()->GetBinLowEdge(bindown); // min bin of the gauss (R2)
+
+  //////////////////////////////////////////////////////////////////////////
+  // left side Gauss pre-fit
+  //////////////////////////////////////////////////////////////////////////
+  cout<<"\nProcessing fit Gauss left..."<< endl;
+  testfit = new TF1("testfit", gaus.data(), 0., 2.);
+  testfit->SetParameter(0, hrsp->GetBinContent(binmax));
+  testfit->SetParameter(1, xbinmax);
+  testfit->FixParameter(1, xbinmax);
+  testfit->SetParameter(2, 0.10);
+  testfit->SetParLimits(2, 0., 1.e+1); 
+  gausLeft = new TF1("gausLeft", gaus.data(), 0., 2.);
+  do {
+    hrsp->Fit(testfit, "Q0", "", minGaus, xbinmax); // Q: quiet mode; 0: do not draw default canvas
+
+    TH1F * hrspXaxis = hXaxis(hrsp, minGaus, xbinmax); // set axis for KS
+    TF1 *fit=hrsp->GetFunction("testfit");
+    fit->SetNpx(hrspXaxis->GetNbinsX());
+    ks=hrspXaxis->KolmogorovTest((TH1D*)fit->GetHistogram()); 
+    chi2 = testfit->GetChisquare()/max(1, testfit->GetNDF()); 
+
+    --bindown;
+    minGaus = hrsp->GetXaxis()->GetBinLowEdge(bindown);
+    if ( chi2 < 5. && ks>0.05) {
+      gausLeft->SetParameter(0, testfit->GetParameter(0));
+      gausLeft->SetParameter(1, testfit->GetParameter(1));
+      gausLeft->SetParameter(2, testfit->GetParameter(2));	  
+    }
+  } while ( chi2 < 5. && minGaus > 0. && ks>0.05 );
```

Comment:
Do you mean while(chi2<5) here? Not >5? Why would you want the chi2 to go up? Also why do you want the ks to go down?